core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -62,7 +62,12 @@ public static Object getMeasureValueBasedOnDataType(String msrValue, DataType da
       case INT:
         return Double.valueOf(msrValue).longValue();
       case LONG:
-        return Long.valueOf(msrValue);
+        int index = msrValue.indexOf(".");
+        if (index != -1) {
+          return Long.valueOf(msrValue.substring(0,index));
+        } else {
+          return Long.valueOf(msrValue);
+        }
```

Comment:
@lion-x ...It is a correct behavior to return null value in this case as our validation is strict. If any column which has BigInt datatype is included as dimension then also we make decimal value as null because of strict validation. 
That is why I suggested that dont change the behavior for Long parsing, instead change the parsing logic for Int datatype to keep the strict validation behavior.